screen-orientation/orientation-reading.html

```diff
@@ -90,8 +90,16 @@
   const type = screen.orientation.type;
   const angle = screen.orientation.angle;
 
-  screen.orientation.type = "foo";
-  screen.orientation.angle = 42;
+  try {
+    screen.orientation.type = "foo";
+  } catch (err) {
```

Comment:
https://webidl.spec.whatwg.org/#dfn-attribute-setter (Step 2) says if readonly, return undefined.
Seems unexpected to me that this would throw.

Reply:
Agree. I don't think this should throw. Trying to assign a value just gets ignored.

Reply:
From https://webidl.spec.whatwg.org/#idl-attributes

> The attribute is read only if the readonly keyword is used before the attribute keyword. An object that implements the interface on which a read only attribute is defined will not allow assignment to that attribute. It is language binding specific whether assignment is simply disallowed by the language, ignored or an exception is thrown.

It is possible to throw an exception for `readonly  attribute`. All browser will throw an exception for this test (https://wpt.fyi/results/screen-orientation/orientation-reading.html?label=experimental&label=master&aligned).

Reply:
Browsers don't know in regular mode. However, they do in strict mode. My bet is that this test is running in strict mode if an exception is thrown.

Reply:
Yeah, it's running in `<script type="module">`, which defaults to strict mode.